crates/store/src/filesystem/utility.rs

```diff
@@ -0,0 +1,44 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: AGPL-3.0-only
+
+use crate::Error;
+
+use cap_async_std::fs::DirEntry;
+use serde::Serialize;
+use sha2::{Digest, Sha256};
+
+pub const ALGORITHM_NAME: &str = "len-sha256";
+
+pub fn len_sha256(buffer: &[u8]) -> String {
```

Comment:
bincode is used for serialization not for hash calculation. The internal hash crate uses `sha2` so I just use the same thing. I don't think the internal hash crate will let me get a single hash in a simple way.

Reply:
trying to get this working now btw. Looks like my existing code is handling it poorly. Might be creating additional directories or something because of how the digest is represented as base64